### PR TITLE
feat: consume profile properties from Okta

### PR DESCRIPTION
**Pull request by DerekTBrown**
This PR consumes profile fields from Okta (as opposed to using email for everything).

### CODE REVIEW

File: plugins/backend/catalog-backend-module-okta/src/providers/OktaUserEntityProvider.ts [hunk 1]
```diff
@@ -78,7 +78,7 @@ export class OktaUserEntityProvider extends OktaEntityProvider {
         },
         spec: {
           profile: {
-            displayName: user.profile.email,
+            displayName: user.profile.displayName,
```
Comment (punkle) [inline plugins/backend/catalog-backend-module-okta/src/providers/OktaUserEntityProvider.ts:81]: Is display guaranteed? Should it default to the email?

Do you think title of the entity should show the display name also?

Comment (DerekTBrown) [reply plugins/backend/catalog-backend-module-okta/src/providers/OktaUserEntityProvider.ts:81]: > Is display guaranteed? 

I believe the `displayName` is guaranteed to be present on the `profile` object:
https://github.com/okta/okta-sdk-nodejs/blob/master/src/types/models/UserProfile.d.ts#L28

My read from this [article](https://support.okta.com/help/s/article/User-display-name-does-not-have-a-value?language=en_US) is that the `displayName` defaults to `firstname lastname`, is user configurable, and will default to empty string if not properly configured. 

> Should it default to the email?

I don't have a strong preference, but out of simplicity I think it is fine for `displayName` to match between Okta and Backstage. Users can configure expressions in Okta to set `displayName` to the email by default if they need.

> Do you think title of the entity should show the display name also?

`displayName` is potentially modifiable by Okta users, and therefore can potentially contain garbage (ex. spaces, emojis, things that aren't identifiable). I think `email` is a better cannonicalization of the user at the moment.

**Review by punkle — DISMISSED** [review]

**Comment by DerekTBrown** [comment]
@kissmikijr I added a changeset. I haven't used this tool before- so let me know if I did everything correctly.

**Comment by kissmikijr** [comment]
@DerekTBrown Looks good thank you! Especially kudos for the nice descriptive changeset message! :) I'll merge the PR after the build passes! thank you for your contribution!

**Review by kissmikijr — APPROVED** [review]

**Comment by madianas21** [comment]
What happened to this PR? Just wondering. I do not see it in the latest [release](https://github.com/DerekTBrown/roadie-backstage-plugins/blob/1b9ba74e729d24d0d61e86f59a514a4635bd8bf2/plugins/backend/catalog-backend-module-okta/src/providers/OktaUserEntityProvider.ts#L81)

**Pull request merged**